### DISCUSSION TITLE
Discord Rich Presence

### DISCUSSION THREAD

**Discussion by StilauGamer** [coder/code-server#5145] [Q&A]
So yeah, as in the title. Discord rich presence.

Is it possible to use an extension on the code-server to show a presence on discord of you coding in the code-server??

( If you wonder why, I just like to show what I do 😂 idk why lol )

**Comment by Qubik65536** [comment]
> So yeah, as in the title. Discord rich presence.
> 
> 
> 
> Is it possible to use an extension on the code-server to show a presence on discord of you coding in the code-server??
> 
> 
> 
> ( If you wonder why, I just like to show what I do 😂 idk why lol )

It is probably not possible because for Discord, you are just using a web browser and browsing a website. Unless Discord have a way to know which web site you are on, Discord is not going to be aware of your are using code-server. As far as I know, Discord is only be able to know which application is running.

**Comment by lg-nineoneninethreenine** [comment]
Well, I don't know. If Discord has an API, not a webhook, this could be possible. But again, IDK.

**Comment by jsjoeio** [comment]
I found these two extensions. I don't know if they'll work with code-server but you can try it!
- https://marketplace.visualstudio.com/items?itemName=icrawl.discord-vscode
- https://marketplace.visualstudio.com/items?itemName=maxerbox.vscode-discord

**Comment by Me0wzz** [comment]
I developed a Discord Rich Presence extension for code-server.

It works entirely as a browser extension. The extension reads the active editor state directly from the code-server tab and publishes the presence to Discord using OAuth PKCE and Discord headless sessions, so no native bridge, IPC, Native Messaging, or other external program is required.

Just install it in the browser you use for code-server, configure it, and it should work right away.

https://github.com/Me0wzz/code-server-discord-presence
